validator/keymanager/remote-web3signer/client.go

```diff
@@ -0,0 +1,177 @@
+package remote_web3signer
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+const (
+	ethApiNamespace = "/api/v1/eth2/sign/"
+	maxTimeout      = 5 * time.Second
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type client struct {
+	BasePath   string
+	restClient HTTPClient
+}
+
+func newClient(endpoint string) (*client, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid Format, unable to parse url")
+	}
+	return &client{
+		BasePath: u.Host,
+		restClient: &http.Client{
+			Timeout: maxTimeout,
+		},
+	}, nil
+}
+
+type SignRequest struct {
+	Type         string        `json:"type"`
+	ForkInfo     *ForkInfo     `json:"fork_info"`
+	SigningRoot  string        `json:"signingRoot"`
+	RandaoReveal *RandaoReveal `json:"randao_reveal"`
+}
+
+type ForkInfo struct {
+	Fork                  *Fork  `json:"fork"`
+	GenesisValidatorsRoot string `json:"genesis_validators_root"`
+}
+
+type Fork struct {
+	PreviousVersion string `json:"previous_version"`
+	CurrentVersion  string `json:"current_version"`
+	Epoch           string `json:"epoch"`
+}
+
+type RandaoReveal struct {
+	Epoch string `json:"epoch"`
+}
+
+type signResponse struct {
+	Signature string `json:"signature"`
+}
+
```

Comment:
// Sign makes an API request to the remote web3signer server and returns the BLS signature.